cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,211 @@ See the Apache™ Twill®
 `API documentation <http://twill.incubator.apache.org/apidocs/org/apache/twill/filesystem/Location.html>`__
 for additional information about the ``Location`` abstraction.
 
+==================
+PartitionedFileSet
+==================
+
+While a FileSet is a convenient abstraction over actual file system interfaces, it still requires
+the application to be aware of file system paths. For example, an application that maintains data
+over time might have a new file for every month. One could come up with a naming convention that encodes
+the month into each file name, and share that convention across all applications that use this file set.
+Yet that can become tedious to manage, especially if the naming convention should ever change—then all
+applications would have to be changed simultaneously for proper functioning.
+
+The PartitionedFileSet Dataset relieves applications from understanding file name conventions. Instead,
+it associates a partition key with every file, for example the year and month associated with that file.
+Because different files cannot have the same partition key, this allows applications to address the
+data uniquely through its partition keys, or more broadly through conditions over the partition keys,
+for example, the months of February through June of a particular year, or the month of November in any
+year. Inheriting the attributes of FileSets—like format and schema—PartitionedFileSets are a powerful
+abstraction over data that is organized into files.
```

Comment:
By inheriting the attributes |---| such as format and schema |---| of FileSets, PartitionedFileSets are a powerful
abstraction over data that is already organized into files. [suggestion: I just a felt a comma was called for.]